engine/src/main/java/org/terasology/rendering/nui/layouts/FlowLayout.java

```diff
@@ -54,14 +62,15 @@ public void onDraw(Canvas canvas) {
         int heightOffset = 0;
         for (UIWidget widget : contents) {
             Vector2i size = canvas.calculatePreferredSize(widget);
-            if (filledWidth != 0 && filledWidth + size.x  > canvas.size().x) {
+            if (filledWidth != 0 && filledWidth + size.x > canvas.size().x) {
                 heightOffset += filledHeight;
                 filledWidth = 0;
                 filledHeight = 0;
             }
             canvas.drawWidget(widget, Rect2i.createFromMinAndSize(filledWidth, heightOffset, size.x, size.y));
             filledWidth += size.x;
             filledHeight = Math.max(filledHeight, size.y);
+            heightOffset += verticalSpacing;
```

Comment:
This adds a height offset after each element, but I think this should only happen in case there is a "wrap", i.e., inside the conditional above.

Reply:
oh indeed, I'll look into adding it only when there is a wrap, and then make a commit for it!